quinn/src/platform/unix.rs

```diff
@@ -296,6 +301,26 @@ fn prepare_msg(
     } else {
         encoder.push(libc::IPPROTO_IPV6, libc::IPV6_TCLASS, ecn);
     }
+
+    if let Some(segment_size) = transmit.segment_size {
+        debug_assert!(
+            U::caps().gso,
+            "Platform must support GSO for setting segment size"
+        );
+
+        #[cfg(target_os = "linux")]
+        fn set_segment_size(encoder: &mut cmsg::Encoder, segment_size: u16) {
+            encoder.push(libc::SOL_UDP, libc::UDP_SEGMENT, segment_size);
```

Comment:
Doesn't this need to check whether GSO is actually supported, rather than running unconditionally on linux?

Reply:
`sendmsg` will check it and return an error if the cmsg is not supported

Reply:
and there is the `debug_assert`.  Could be a real assert or an error too. But I think leaving it out in release mode is ok, since this is definitely a config issue.

Reply:
Won't that fail the whole operation?

Reply:
Oh, right, this is all guarded on `transmit.segment_size` to begin with.